control-plane/agents/src/bin/core/volume/specs.rs

```diff
@@ -745,12 +750,18 @@ impl ResourceSpecsLocked {
     }
 
     /// Get or Create the resourced VolumeSpec for the given request.
-    pub(crate) fn get_or_create_volume(&self, request: &CreateVolume) -> ResourceMutex<VolumeSpec> {
+    pub(crate) fn get_or_create_volume(
+        &self,
+        request: &CreateVolume,
+        content_source: Option<CloneContentSource>,
```

Comment:
this is very confusing, let's not do it this way, we can either have different methods or the clone impl can simply modify the created spec?